addon/components/pie-chart.js

```diff
@@ -483,9 +483,17 @@ export default ChartComponent.extend(FloatingTooltipMixin,
     groups.select('path').attr(this.get('sliceAttrs'));
     
     var labelWidth = this.get('labelWidth');
+    var maxLabelWidth = this.get('outerWidth')/2 - this.get('labelPadding');
     var labelTrimmer = LabelTrimmer.create({
-      getLabelSize: function() {
-        return labelWidth;
+      getLabelSize: function(d, selection) {
+        var text, xPos;
+        // Pick the current text element by filtering on data
+        text = selection.filter(function(data, i) {
+          if (data === d) return this;
```

Comment:
I don't understand what's going on here. The callback passed to `filter` just needs to return a boolean